packages/strapi-database/lib/database-manager.js

```diff
@@ -16,6 +16,59 @@ class DatabaseManager {
     this.models = new Map();
   }
 
+  // Check if all collection names are unique
+  checkForDuplicates() {
+    const createErrorMessage = (
+      collectionA,
+      collectionB
+    ) => `Duplicated collection name: \`${collectionA.collectionName}\`.
+The same collection name can't be used for two different models.
+First found in ${collectionA.origin} \`${collectionA.apiOrPluginName}\`, model \`${collectionA.modelName}\`.
+Second found in ${collectionB.origin} \`${collectionB.apiOrPluginName}\`, model \`${collectionB.modelName}\`.`;
+
+    const collections = [];
+    _.forIn(this.strapi.admin.models, (model, modelName) => {
+      collections.push({
+        origin: 'Strapi internal',
+        collectionName: model.collectionName || `${modelName}`.toLocaleLowerCase(),
```

Comment:
I think the collectionName is always declared on the model when this code executes.